hudi-common/src/main/java/org/apache/hudi/keygen/KeyGenerator.java

```diff
@@ -56,4 +61,35 @@ public List<String> getRecordKeyFieldNames() {
     throw new UnsupportedOperationException("Bootstrap not supported for key generator. "
         + "Please override this method in your custom key generator.");
   }
+
+  public static String constructRecordKey(List<String> recordKeyFields, List<Object> recordKeyValues) {
+    if (recordKeyFields.size() == 1) {
```

Comment:
We could also possibly get rid of this check by having another `constructRecordKey` method on a single field since the reader context knows the number of fields during initialization, so this check only lives in the constructor of the reader context.

Reply:
This is feeling like a large micro optimization, with branch prediction this may not be that big of a win and this is for a case we're considering deprecating (virtual keys)

Reply:
Sg.  OK punting this.